File: webflux-dialogue/src/main/java/com/study/webflux/rag/infrastructure/mission/repository/UserMissionMongoRepository.java
package com.study.webflux.rag.infrastructure.mission.repository;

import org.springframework.data.mongodb.repository.ReactiveMongoRepository;

import com.study.webflux.rag.infrastructure.mission.document.UserMissionDocument;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface UserMissionMongoRepository
	extends ReactiveMongoRepository<UserMissionDocument, String> {

	Flux<UserMissionDocument> findByUserId(String userId);

	Mono<UserMissionDocument> findByUserIdAndMissionId(String userId, String missionId);
}

File: backend/infrastructure/src/main/kotlin/com/miyou/app/infrastructure/mission/repository/UserMissionMongoRepository.kt
package com.miyou.app.infrastructure.mission.repository

import com.miyou.app.infrastructure.mission.document.UserMissionDocument
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface UserMissionMongoRepository : ReactiveMongoRepository<UserMissionDocument, String> {
    fun findByUserId(userId: String): Flux<UserMissionDocument>

    fun findByUserIdAndMissionId(
        userId: String,
        missionId: String,
    ): Mono<UserMissionDocument>
}
